documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1837,3 +1838,19 @@ or
 ```
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="r4004" ></a>R4004 OperationIdRequired
+
+**Category** : ARM Warning
+
+**Applies to** : ARM OpenAPI(swagger) specs
+
+**Output Message** : Missing operationId , path '${operation path}', method '${operation method}'  
+
+**Description** : Each operation must has a operationId
+
+**Why this rule is important**: OperationId is required in each operations
```

Comment:
for "why", how about to describe the consequence if not?  I am fine with others. Thanks.